docs/building-a-form/common-patterns-for-building-forms.md

```diff
@@ -151,64 +151,52 @@ To create a block of text with no fields that follow, create an empty view objec
-  "type": "object",
-  "properties": {
-    "disability": { "type": "boolean" },
-    "dic": { "type": "boolean" },
-    "chapter31": { "type": "boolean" },
-    "ownServiceBenefits": { "type": "string" },
-    "chapter35": { "type": "boolean" },
-    "chapter33": { "type": "boolean" },
-    "transferOfEntitlement": { "type": "boolean" },
-    "other": { "type": "string" },
-    "veteranFullName": { "$ref": "#/definitions/fullName" },
-    "veteranSocialSecurityNumber": { "$ref": "#/definitions/ssn" }
+schema: {
+  type: 'object',
+  properties: {
+    employed: { type: 'boolean' },
+    jobStartDate: { type: 'string' },
+    monthlyWages: { type: 'string' },
+    otherMonthlyIncome: { type: 'string' }
   }
 }
+
 ```
 
-Only `chapter35`, `chapter33`, `transferOfEntitlement`, `veteranFullName`, and `veteranSocialSecurityNumber` are conditionally hidden, so the `schema` and `uiSchema` are written as:
+To accomplish this, the `schema` and `uiSchema` are written as:
 
 ```js
-// schema
-{
-  disability: { ... },
-  dic: { ... },
-  chapter31: { ... },
-  ownServiceBenefits: { ... },
-  'view:sponsorServiceOptions': {
-    chapter35: { ... },
-    chapter33: { ... },
-    transferOfEntitlement: { ... },
-    veteranFullName: { ... },
-    veteranSocialSecurityNumber: { ... }
-  },
-  other: { ... }
-}
-
-// uiSchema
-{
-  disability: { ... },
-  dic: { ... },
-  chapter31: { ... },
-  ownServiceBenefits: { ... },
-  'view:sponsorServiceOptions': {
-    hideIf: (formData) => /* Some condition here */,
-    chapter35: { ... },
-    chapter33: { ... },
-    transferOfEntitlement: { ... },
-    veteranFullName: { ... },
-    veteranSocialSecurityNumber: { ... }
+schema: {
+  type: 'object',
+  properties: {
+    employed: { type: 'boolean' },
+    'view:jobInformation': {
+      type: 'object',
+      properties: {
+        jobStartDate: { type: 'string' },
+        monthlyWages: { type: 'string' }
+      }
+    },
+    otherMonthlyIncome: { type: 'string' }
+  }
+},
+uiSchema: {
+  employed: { 'ui:title': 'I am employed' },
+  'view:jobInformation': {
+    'ui:options': {
+      hideIf: (formData) => !formData.employed
+    },
+    jobStartDate: { 'ui:title': 'Job start date' },
+    monthlyWages: { 'ui:title': 'Monthly wages' },
   },
-  other: { ... }
+  otherMonthlyIncome: { 'ui:title': 'Other monthly income' }
 }
 ```
 
-From this, the fields in the `view:sponsorServiceOptions` object are moved up one level and sent alongside `dic` and `chapter31`. The back end doesn't see objects with names that start with `view:`, but it gets all fields inside those objects.
+The back end doesn't see objects with names that start with `view:`, but it gets all fields inside those objects because the form processing includes them in the parent object (in this example, the one containing `employed` and `otherMonthlyIncome`). The `hideIf` function is passed a copy of the current `formData`, in this case it will hide the fields unless `employed` is `true`.
```

Comment:
The wording here was a bit confusing to me, perhaps this is clearer?

"The `hideIf` function is passed a copy of the current `formData` in order to determine the condition upon which the fields are shown. In this example, it will hide the fields unless `employed` is `true`.

Objects from the form config with names that start with `view:` are not passed to the back-end, but fields within those `view:` objects are passed to the back-end by including them in the parent object. In this example, the fields `jobStartDate` and `monthlyWages` would be included in the same object as `employed` and `otherMonthlyIncome`, while the field for `view:JobInformation` would be filtered out."

One thing we may want to answer is whether fields that are conditionally shown are always included in the final data object that is submitted to the back-end, or only included if they are shown. Looking at the code it seems like fields that are conditionally shown are included; I don't see any logic that filters those out. It's only complete pages that are filtered out. But it seems like it would make more sense if fields that are never shown to the user are also filtered out from the final submission? I need to do more digging into this.